### DISCUSSION TITLE
Override file base routing

### DISCUSSION THREAD

**Discussion by PizzaConsole** [vuejs/vitepress#3840] [Q&A]
Would it be possible to create a plugin that can override the file based routing? Where in the `.md` file you could include some thing like 

```
---
url: /different/path
---
```
and have that be used to build the html file?

**Comment by brc-dd** [comment]
https://vitepress.dev/guide/routing#route-rewrites

if you want to keep it in frontmatter, you can do something like https://github.com/vuejs/vitepress/issues/285#issuecomment-1722254567

**Reply by PizzaConsole** [reply]
@brc-dd Thanks for the fast response! I had read the rewrites section, but it seemed tedious with what I was going for. Though your example does seem to open up some possibilities that I had not considered! I will play around with ii and see what I can get working

**Reply by PizzaConsole** [reply]
@brc-dd  Is it possible to get [createcontentloader ](https://vitepress.dev/guide/data-loading#createcontentloader) to execute after the rewrite? When I use it, the `url` returned in the `createcontentloader` is the `url` prior to the rewrite.
